modules/lighthouse_service.js

```diff
@@ -118,13 +118,124 @@ const createReport = async () => {
   }
 }
 
+// Get Robots
+const getRobots = async () => {
+  try {
+    const response = await axios.get(
+      `${config.privateRuntimeConfig.lighthouseBaseURL}/beckman/robots`
+    )
+    return {
+      success: true,
+      robots: response.data.robots
+    }
+  } catch (resp) {
+    const errors = resp.response
+      ? resp.response.data.errors
+      : [resp.message + ': Failed to get Robots from Lighthouse Service']
+    return {
+      success: false,
+      errors
+    }
+  }
+}
+
+// Get Failure Types
+const getFailureTypes = async () => {
+  try {
+    const response = await axios.get(
+      `${config.privateRuntimeConfig.lighthouseBaseURL}/beckman/failure-types`
+    )
+    return {
+      success: true,
+      failure_types: response.data.failure_types
+    }
+  } catch (resp) {
+    const errors = resp.response
+      ? resp.response.data.errors
+      : [resp.message + ': Failed to get Failure Types from Lighthouse Service']
+    return {
+      success: false,
+      errors
+    }
+  }
+}
+
+// Create Destination Plate
+// Returned on success:
+//  { success: true, response: "A successful message" }
+// Returned on failure:
+//  { success: false, errors: ["A failure message"] }
+const createDestinationPlate = async (username, barcode, robotSerialNumber) => {
+  try {
+    const response = await axios.get(
+      `${config.privateRuntimeConfig.lighthouseBaseURL}/cherrypicked-plates/create?barcode=${barcode}&robot=${robotSerialNumber}&user_id=${username}`
+    )
+    const responseData = response.data.data
+    // success
+    return {
+      success: true,
+      response: `Successfully created destination plate, with barcode: ${responseData.plate_barcode}, and ${responseData.number_of_positives} positive sample(s)`
+    }
+  } catch (resp) {
+    const errors = resp.response.data
+    // failure
+    return {
+      success: false,
+      ...errors
+    }
+  }
+}
+
+// Fail Destination Plate
+// Returned on success:
+//  { success: true, errors: [] }
+// Returned on partial success:
+//  { success: true, errors: ["A successful error message"] }
+// Returned on failure:
+//  { success: false, errors: ["A failure message"] }
+const failDestinationPlate = async (
+  username,
+  barcode,
+  robotSerialNumber,
+  failureType
+) => {
+  try {
+    const response = await axios.get(
+      `${config.privateRuntimeConfig.lighthouseBaseURL}/cherrypicked-plates/fail?barcode=${barcode}&robot=${robotSerialNumber}&user_id=${username}&failure_type=${failureType}`
+    )
+    // partial success
```

Comment:
Is this a success or is it a failure? You can always throw an error if it is a failure. I am still confused.

Reply:
@Chris-Friend explained it as: _"I've sent your message, but be aware that..."._  So the message has still successfully been send to the MLWH, so its a 200. But the UI is showing this response with the alert type as 'warning' to make the user aware